kubetest/anywhere.go

```diff
@@ -34,6 +34,7 @@ var (
 	kubernetesAnywhereKubeadmVersion = flag.String("kubernetes-anywhere-kubeadm-version", "stable", "(kubernetes-anywhere only) Version of kubeadm to use, if phase2-provider is kubeadm. May be \"stable\" or a gs:// link to a custom build.")
 	kubernetesAnywhereCluster        = flag.String("kubernetes-anywhere-cluster", "", "(kubernetes-anywhere only) Cluster name. Must be set for kubernetes-anywhere.")
 	kubernetesAnywhereUpTimeout      = flag.Duration("kubernetes-anywhere-up-timeout", 20*time.Minute, "(kubernetes-anywhere only) Time limit between starting a cluster and making a successful call to the Kubernetes API.")
+	kubernetesAnywhereCNIName        = flag.String("kubernetes-anywhere-cni-name", "weave", "(kubernetes-anywhere only) The name of the CNI plugin used for the cluster's SDN.")
```

Comment:
I think I understand your desire to default to `weave` here for backwards compatibility, but the default `phase2-provider` is actually `ignition` (which hardcodes its use of `kubenet` in kubernetes-anywhere), so the default CNI should probably be empty to disable it entirely and not break `ignition` support.

What you could do is add a little extra logic to default to `weave` if the CNI is empty _and_ `phase2-provider=kubeadm`, which should give you stronger backwards-compatibility. Or, you can update all of our existing kubeadm e2e jobs to make sure they specify the new flag (and don't break), but that would take more work and coordination, and I am a big fan of backwards-compatibility for smooth, non-atomic transitions.